zone/client.cpp

```diff
@@ -10129,3 +10129,50 @@ void Client::SetAFK(uint8 afk_flag) {
 	entity_list.QueueClients(this, outapp);
 	safe_delete(outapp);
 }
+
+void Client::SendToInstance(std::string instance_type, std::string zone_short_name, uint32 instance_version, float x, float y, float z, float heading, std::string instance_identifier, uint32 duration) {
+	uint32 account_id = AccountID();
+	std::string account_name = AccountName();
+	uint32 zone_id = ZoneID(zone_short_name);
+	std::string current_instance_type = str_tolower(instance_type);
+	std::string instance_type_name = "public";
+	if (current_instance_type.find("solo") != std::string::npos) {
+		instance_type_name = GetCleanName();
+	} else if (current_instance_type.find("group") != std::string::npos) {
+		uint32 group_id = (GetGroup() ? GetGroup()->GetID() : 0);
+		instance_type_name = itoa(group_id);
+	} else if (current_instance_type.find("raid") != std::string::npos) {
+		uint32 raid_id = (GetRaid() ? GetRaid()->GetID() : 0);
+		instance_type_name = itoa(raid_id);
+	} else if (current_instance_type.find("guild") != std::string::npos) {
+		uint32 guild_id = (GuildID() > 0 ? GuildID() : 0);
+		instance_type_name = itoa(guild_id);
+	}
+
+	std::string full_bucket_name = fmt::format(
+		"{}_{}_{}",
+		instance_type_name,
```

Comment:
`current_instance_type` should be included in the naming of this key not only for debugging / identifying purposes but it is possible for each of these different types of ID's to collide with each other